src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo.cs

```diff
@@ -105,9 +117,6 @@ private Task<IEnumerable<ISymbol>> FindAsyncWorker(SearchQuery query, AsyncLazy<
                     return this.FindAsync(lazyAssembly, query.Name, ignoreCase: true, cancellationToken: cancellationToken);
                 case SearchKind.Fuzzy:
                     return this.FuzzyFindAsync(lazyAssembly, query.Name, cancellationToken);
-                case SearchKind.Custom:
```

Comment:
This is trivial to prove without having to even see all the codepaths that flowed here.  Here's why:

SearchQuery.Kind was never set to Custom in any circumstance.  Even if a custom predicate was supplied, the Kind of the query was still set to .Exact.  As such, even if we were called, we would have gone into the ```case SearchKind.Exact:``` case.

Now, i actually did the work to walk backward.  The only code that flowed here was from the Add-Import codebase, and a few SymbolFinder public entry points.  However, every single entry point that flowed here only made queries with the kind 'Exact' 'ExactIgnoreCase' and 'Custom'. 

So this code is totally safe to delete.  I've also added asserts to ensure that we don't accidenly pass through a custom search query here in the future.